### PR TITLE
Resolves #288, Store Asciidoctor::Document object in page context

### LINKED ISSUE #288: Store Asciidoctor::Document object in page context
When rendering an AsciiDoc document using Asciidoctor, Awestruct should expose the Asciidoctor::Document object via the page context. This change will allow the layout to access information from the document for populating the sidebar, or other portions of the layout, with information such as the table of contents, list of links, list of figures, etc.

### PR DESCRIPTION
**Pull request by Mogztter**
Store `Asciidoctor::Document` object in page context

Resolves #288

### CODE REVIEW

File: lib/awestruct/handlers/template/asciidoc.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def prepare
       end
 
       def evaluate(scope, locals, &block)
-        @output ||= ::Asciidoctor.render(data, options, &block)
+        @output ||= (scope.document = ::Asciidoctor.load(data, options, &block)).convert
```
Comment (Mogztter) [inline lib/awestruct/handlers/template/asciidoc.rb]: @mojavelinux Not sur about `&block` parameter ?

Comment (mojavelinux) [inline lib/awestruct/handlers/template/asciidoc.rb]: The &block parameter is ignored by Asciidoctor, so there's no need to pass
it on.

**Comment by LightGuard** [comment]
Thanks!

**Comment by i386** [comment]
@Mogztter this is pretty useful - any chance we could get this in a release? :)

**Pull request merged**